docs/versioned_docs/version-v1.2.0/02-Features/message-pumps/customization.md

```diff
@@ -0,0 +1,269 @@
+---
+title: "Message Pump Customization"
+layout: default
+---
+
+# Customize message pumps
+
+While the message processing is handled by the `IMessageHandler<>` implementations, the message router controls in what format the message is received.
+We allow several customizations while using a built-in or implementing your own message router.
+
+- [Customize message pumps](#customize-message-pumps)
+  - [Filter messages based on message context](#filter-messages-based-on-message-context)
+  - [Control custom deserialization](#control-custom-deserialization)
+    - [Extending the message router](#extending-the-message-router)
+    - [Bring your own deserialization](#bring-your-own-deserialization)
+  - [Filter messages based on message body](#filter-messages-based-on-message-body)
+  - [Fallback message handling](#fallback-message-handling)
+
+## Filter messages based on message context
+
+When registering a new message handler, one can opt-in to add a filter on the message context which filters out messages that are not needed to be processed.
+
+This can be useful when you are sending different message types on the same queue. Another use-case is being able to handle different versions of the same message type which have different contracts because you are migrating your application.
+
+Following example shows how a message handler should only process a certain message when a property's in the context is present.
+
+We'll use a simple message handler implementation:
+
+```csharp
+using Arcus.Messaging.Abstractions;
+using Arcus.Messaging.Pumps.Abstractions.MessagingHandling;
+
+public class OrderMessageHandler : IMessageHandler<Order>
+{
+    public async Task ProcessMessageAsync(Order order, MessageContext context, ...)
+    {
+        // Do some processing...
+	}
```

Comment:
nitpicking, but this closing brace should be correctly indented :)

Reply:
Ah, it was bc it is a tab instead of spaces. I'll update this on all versions. 👍